docs/structured-streaming-kafka-integration.md

```diff
@@ -187,50 +303,68 @@ The following options must be set for the Kafka source.
 The following configurations are optional:
 
 <table class="table">
-<tr><th>Option</th><th>value</th><th>default</th><th>meaning</th></tr>
+<tr><th>Option</th><th>value</th><th>default</th><th>mode</th><th>meaning</th></tr>
 <tr>
   <td>startingOffsets</td>
-  <td>earliest, latest, or json string
+  <td>earliest, latest (streaming only), or json string
   {"topicA":{"0":23,"1":-1},"topicB":{"0":-2}}
   </td>
-  <td>latest</td>
+  <td>streaming=latest, batch=earliest</td>
+  <td>streaming and batch</td>
   <td>The start point when a query is started, either "earliest" which is from the earliest offsets,
   "latest" which is just from the latest offsets, or a json string specifying a starting offset for
   each TopicPartition.  In the json, -2 as an offset can be used to refer to earliest, -1 to latest.
-  Note: This only applies when a new Streaming query is started, and that resuming will always pick
-  up from where the query left off. Newly discovered partitions during a query will start at
+  Note: For Batch, latest (either implicitly or by using -1 in json) is not allowed.
+  For Streaming, this only applies when a new Streaming query is started, and that resuming will
+  always pick up from where the query left off. Newly discovered partitions during a query will start at
   earliest.</td>
 </tr>
 <tr>
+  <td>endingOffsets</td>
+  <td>latest or json string
+  {"topicA":{"0":23,"1":-1},"topicB":{"0":-1}}
+  </td>
+  <td>latest</td>
+  <td>batch only</td>
+  <td>The end point when a batch query is started, either "latest" which is just from the latest
```

Comment:
nit: a json string specifying **a starting** offset for each TopicPartition. -> a json string specifying **an ending** offset for each TopicPartition.